components/drawer/index.tsx

```diff
@@ -82,6 +84,9 @@ const Drawer: React.FC<DrawerProps> & {
     ...rest
   } = props;
 
+  let ariaId: string | undefined = useId();
+  ariaId = rest.title ? ariaId : undefined;
```

Comment:
issue: if someone add a `aria-label="Something"`  to the drawer component this mess everything up 

We need to make sure to add the ariaId only if needed ! 

Another problem because of this PR is that in tests the useId always return `test-id` and if you have more than one component in your test that use useId then you have a bunch of conflicts !

```
const notificationSettingsDrawer = screen.getByRole('dialog', {
        name: 'Notification Settings',   
      })
```
wont work anymore if other component use  useId somewhere else  (for example if one drawer open another drawer !

Reply:
Workaround

```
    <Drawer
      aria-label="DONT IGNORE ME"
      aria-labelledby='' <------- add this to skip this PR code 
      title="Dialog title"
```